java.hints/src/org/netbeans/modules/java/hints/bugs/NPECheck.java

```diff
@@ -1046,19 +1066,40 @@ public State visitIdentifier(IdentifierTree node, Void p) {
             
             Element e = info.getTrees().getElement(getCurrentPath());
 
-            if (e == null || !isVariableElement(e)) {
+            if (!Utilities.isValidElement(e)) {
                 return State.POSSIBLE_NULL;
             }
-            if (e.getKind() == ElementKind.ENUM_CONSTANT) {
-                // enum constants are never null
-                return State.NOT_NULL;
+            switch (e.getKind()) {
+                case ENUM_CONSTANT:
+                    // enum constants are never null
+                    return State.NOT_NULL;
+                case FIELD:
+                    Set<Modifier> modifiers = e.getModifiers();
+                    // assume that any static final field of a standard class
+                    // in the java.** or javax.** packages is not null.
+                    // This is specifically to make sure that the state of
+                    // TRUE and FALSE static imports from java.lang.Boolean
+                    // is NOT_NULL.
+                    if (modifiers != null && modifiers.contains(Modifier.STATIC) && modifiers.contains(Modifier.FINAL)) {
```

Comment:
I would be cautious to add rather broad-scope heuristics like this. It is rather necessary for the fix to work OK, but consider if some more restrictive condition would not be acceptable - e.g. just static final fields of primitive wrappers, or java.lang.* package. 

With such rather broad scope, I'd like to see an UI option introduced - something like list of REs the FQN is matched against.

@jlahoda  may have an opinion on this, too